<?xml version="1.0" ?>
<robot name="test_robot">
   <!-- Color of materials -->
  <material name="grey">
    <color rgba="0.2 0.2 0.2 1.0"/>
  </material>
  <material name="yellow">
    <color rgba="1 1 0 1"/>
  </material>
  <material name="blue">
    <color rgba="0 0 1 1"/>
  </material>
  <material name="red">
    <color rgba="1 0 0 1"/>
  </material>
  <material name="green">
    <color rgba="0 1 0 1"/>
  </material>
    <material name="magenta">
    <color rgba="1 0 1 1"/>
  </material>
  <!-- Base -->
  <link name="base_link">
    <visual>
      <geometry>
        <box size="0.07 0.07 0.07"/>
      </geometry>
      <origin rpy="0 0 0" xyz="-0.035 0 0.035"/>
      <material name="blue" />
    </visual>
  	<collision>
      <geometry>
        <box size="0.07 0.07 0.07"/>
      </geometry>
    </collision>
  	<inertial>
      <mass value="1"/>
      <inertia ixx="0.082" ixy="0.0" ixz="0.0" iyy="0.082" iyz="0.0" izz="0.082"/>
    </inertial>
  </link>
 <link name="link1">
    <visual>
      <geometry>
        <box size="0.02 0.02 0.02"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0.009 0 0"/>
      <material name="yellow" />
    </visual>
	<collision>
      <geometry>
        <box size="0.02 0.02 0.02"/>
      </geometry>
    </collision>
	<inertial>
      <mass value="0.028"/>
      <inertia ixx="0.00000187" ixy="0.0" ixz="0.0" iyy="0.00000187" iyz="0.0" izz="0.00000187"/>
    </inertial>
  </link>
 <link name="link2">
    <visual>
      <geometry>
        <cylinder length="0.105" radius="0.01"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0.0525"/>
      <material name="green" />
    </visual>
	<collision>
      <geometry>
        <cylinder length="0.105" radius="0.01"/>
      </geometry>
    </collision>
	<inertial>
      <mass value="0.5"/>
      <inertia ixx="0.000472" ixy="0.0" ixz="0.0" iyy="0.000472" iyz="0.0" izz="0.00000417"/>
    </inertial>
  </link>
<link name="link3">
    <visual>
      <geometry>
        <cylinder length="0.097" radius="0.01"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0.052"/>
      <material name="red" />
    </visual>
	<collision>
      <geometry>
        <cylinder length="0.097" radius="0.01"/>
      </geometry>
    </collision>
	<inertial>
      <mass value="0.46"/>
      <inertia ixx="0.000372" ixy="0.0" ixz="0.0" iyy="0.000372" iyz="0.0" izz="0.00000383"/>
    </inertial>
  </link>
  <link name="link4">
    <visual>
      <geometry>
        <cylinder length="0.03" radius="0.01"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0.015"/>
      <material name="magenta" />
    </visual>
	<collision>
      <geometry>
        <cylinder length="0.03" radius="0.01"/>
      </geometry>
    </collision>
	<inertial>
      <mass value="0.285"/>
      <inertia ixx="0.0000855" ixy="0.0" ixz="0.0" iyy="0" iyz="0.0" izz="0.0000855"/>
    </inertial>
  </link>

<link name="link5">
    <visual>
      <geometry>
        <cylinder length="0.160" radius="0.01"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0.08"/>
      <material name="grey" />
    </visual>
	<collision>
      <geometry>
        <cylinder length="0.160" radius="0.01"/>
      </geometry>
    </collision>
	<inertial>
      <mass value="0.76"/>
      <inertia ixx="0.000648" ixy="0.0" ixz="0.0" iyy="0" iyz="0.0" izz="0.000648"/>
    </inertial>
  </link>


  <joint name="joint1" type="revolute">
    <parent link="base_link"/>
    <child link="link1"/>
    <origin xyz="-0.035 0 0.09" rpy="0 1.5 0" />
    <limit effort="300" velocity="0.1" lower="-3.1" upper="3.1"/>
  </joint>

  <joint name="joint2" type="revolute">
    <parent link="link1"/>
    <child link="link2"/>
    <origin xyz="0.026 0 0.01" rpy="0 -1.5 0" />
    <limit effort="300" velocity="0.1" lower="-1.480" upper="1.480"/>
  </joint>

  <joint name="joint3" type="revolute">
    <parent link="link2"/>
    <child link="link3"/>
    <origin xyz="0 0 0.1" rpy="0 0 0" />
    <limit effort="300" velocity="0.1" lower="-1.9" upper="1.9"/>
  </joint>

<joint name="joint4" type="continuous">
    <parent link="link3"/>
    <child link="link4"/>
    <origin xyz="0 0 0.105" rpy="0 1.5 0" />
  </joint>

<joint name="joint5" type="revolute">
    <parent link="link4"/>
    <child link="link5"/>
    <origin xyz="0 0 0.03" rpy="0 0 1.5" />
  <limit effort="300" velocity="0.1" lower="-1.5" upper="1.5"/>
  </joint>

</robot>


